Fix webgestalt missing organism

fischer-hub opened the pull request:
Fix an issue where the webgestaltR function will always return `missing organims` even when the organism variable is set to a valid value. (This was just an outdated package issue I updated webgestaltr from 0.4.4 to 0.4.6)

Fix a new issue with the new version where webgestalt will crash during the report creation because `argument 'zip' must be a non-empty character string` when run with the conda option of the pipeline.

Turns out the zip function in R uses the environment variable `R_ZIPCMD` to determine the system call for zipping things, however per default this variable is empty in conda environment. (Maybe it is empty because we use this old R CMD BATCH call instead of Rscript..?). Anyway the variable is now defined in the nextflow process before running the webgestalt R script which fixed the issue.

Webgestalt now has a separate environment file and resources instead of the ones from the deseq2 process. 
However, its still running on the old docker container for `-profile docker` runs. A new docker container still would have to be added but the environment variable solution should still work with this.

hoelzer commented:
or: I can also build a separate container with these dependencies:

```bash
name: webgestalt
channels:
  - conda-forge
  - bioconda
  - defaults
dependencies:
  - r-base
  - r-webgestaltr=0.4.6
```

fischer-hub commented:
> or: I can also build a separate container with these dependencies:
> 
> ```bash
> name: webgestalt
> channels:
>   - conda-forge
>   - bioconda
>   - defaults
> dependencies:
>   - r-base
>   - r-webgestaltr=0.4.6
> ```

yes that is a good idea i think. I will add the new container to the config then and commit the changes to this PR .

hoelzer commented:
`docker pull nanozoo/webgestalt:0.4.6--e94366e`

fischer-hub commented:
> nanozoo/webgestalt:0.4.6--e94366e

Hm this is not working because the zip command is unknown in the docker container.
The base image of the container probably doesn't come with the zip command right? I added zip to the conda env file so we always have the command available. Could you please rebuild the container with the updated environment file or just add zip to it @hoelzer ? whatever is easier

thx!

hoelzer commented:
Ahhh, yeah the base image is quite minimal to keep the containers small. I added zip! 

`docker pull nanozoo/webgestalt:0.4.6--5a8d95e`

fischer-hub commented:
> Ahhh, yeah the base image is quite minimal to keep the containers small. I added zip!
> 
> `docker pull nanozoo/webgestalt:0.4.6--5a8d95e`

great, I updated the container in the config file it works for me now!

hoelzer commented:
Great, works, thx @fischer-hub !
